client/rest/handlers.go

```diff
@@ -0,0 +1,213 @@
+	if err != nil {
+		writeError(w, err)
+		return
+	}
+	writeSuccess(w, keys)
+}
+
+var (
+	errNonMatchingPathAndJSONKeyNames = errors.New("path and json key names don't match")
+)
+
+func (k *Keys) UpdateKey(w http.ResponseWriter, r *http.Request) {
+	uReq := new(UpdateKeyRequest)
+	if err := parseRequestJSON(r, uReq); err != nil {
+		writeError(w, err)
+		return
+	}
+
+	query := mux.Vars(r)
+	name := query["name"]
+	if name != uReq.Name {
+		writeError(w, errNonMatchingPathAndJSONKeyNames)
+		return
+	}
+
+	if err := k.manager.Update(uReq.Name, uReq.OldPass, uReq.NewPass); err != nil {
+		writeError(w, err)
+		return
+	}
+
+	key, err := k.manager.Get(uReq.Name)
+	if err != nil {
+		writeError(w, err)
+		return
+	}
+	writeSuccess(w, &key)
+}
+
+func (k *Keys) DeleteKey(w http.ResponseWriter, r *http.Request) {
+	dReq := new(DeleteKeyRequest)
+	if err := parseRequestJSON(r, dReq); err != nil {
+		writeError(w, err)
+		return
+	}
+
+	query := mux.Vars(r)
+	name := query["name"]
+	if name != dReq.Name {
+		writeError(w, errNonMatchingPathAndJSONKeyNames)
+		return
+	}
+
+	if err := k.manager.Delete(dReq.Name, dReq.Passphrase); err != nil {
+		writeError(w, err)
+		return
+	}
+
+	resp := &ErrorResponse{Success: true}
+	writeSuccess(w, resp)
+}
+
+func (k *Keys) Register(r *mux.Router) {
+	r.HandleFunc("/keys", k.GenerateKey).Methods("POST")
+	r.HandleFunc("/keys", k.ListKeys).Methods("GET")
+	r.HandleFunc("/keys/{name}", k.GetKey).Methods("GET")
+	r.HandleFunc("/keys/{name}", k.UpdateKey).Methods("POST", "PUT")
+	r.HandleFunc("/keys/{name}", k.DeleteKey).Methods("DELETE")
+}
+
+type Context struct {
+	Keys *Keys
+}
+
+func (ctx *Context) RegisterHandlers(r *mux.Router) error {
+	ctx.Keys.Register(r)
+	r.HandleFunc("/build/send", doSend).Methods("POST")
+	r.HandleFunc("/sign", doSign).Methods("POST")
+
+	return nil
+}
+
+func doSign(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+
+	sr := new(SignRequest)
+	if err := parseRequestJSON(r, sr); err != nil {
+		writeError(w, err)
+		return
+	}
+
+	tx := sr.Tx
+	if err := SignTx(sr.Name, sr.Password, tx); err != nil {
+		writeError(w, err)
+		return
+	}
+	writeSuccess(w, tx)
+}
+
+func doSend(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+	si := new(SendInput)
+	if err := parseRequestJSON(r, si); err != nil {
+		writeError(w, err)
+		return
+	}
+
+	var errsList []string
+	if si.From == nil {
+		errsList = append(errsList, `"from" cannot be nil`)
+	}
+	if si.Sequence <= 0 {
+		errsList = append(errsList, `"sequence" must be > 0`)
+	}
+	if si.To == nil {
+		errsList = append(errsList, `"to" cannot be nil`)
+	}
+	if si.Fees == nil {
+		errsList = append(errsList, `"fees" cannot be nil`)
+	}
+	if len(errsList) > 0 {
+		err := &ErrorResponse{
+			Error: strings.Join(errsList, ", "),
+			Code:  406,
+		}
+		writeCode(w, err, 406)
+		return
+	}
+
+	var tx basecoin.Tx
+	coins := []coin.Coin{*si.Fees}
+	if !si.Multi { // Just doing one transaction
```

Comment:
Ooh I see, thanks for letting me know. I'll do another iteration and work on fixing it up.